primitives/version/src/lib.rs

```diff
@@ -148,6 +149,98 @@ macro_rules! create_apis_vec {
 	};
 }
 
+/// Runtime Version with support for reading
+/// previous runtime version encoding.
+pub struct RuntimeVersionCompatibility(RuntimeVersion);
+
+impl From<RuntimeVersionCompatibility> for RuntimeVersion {
+	fn from(x: RuntimeVersionCompatibility) -> Self {
+		x.0
+	}
+}
```

Comment:
Ahh fuck, encoding/decoding this like an enum doesn't work. I mean it could work, but we can not sure that we not just decoded garbage...

Nevertheless, I would go with the enum and document the behavior that `Core` is bumped when we create a new version.

Reply:
Or we keep just one `struct RuntimeVersion` that is always the latest and then write a custom decode.

After we have decoded the apis, we know how many fields will follow.

Reply:
Using the struct instead of the enum proposed by me above, will also keep the changeset smaller.

Reply:
The issue is that I need to use the old struct for primitive/api/src/lib.rs core runtime api.
Did rewrite decode to skip replay though.